### PR TITLE
fix #17275

### LINKED ISSUE #17275: parseJson big number more than safe number in JS backend
```nim
import std/json

let x = parseJson("9999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999")

echo x.kind
```

```
Expected: JString
Actual: Jint
```

### PR DESCRIPTION
**Pull request by ringabout**
fix #17275

choose safe number as a reasonable dividing line.

### CODE REVIEW

File: lib/pure/json.nim [hunk 1]
```diff
@@ -913,14 +913,20 @@ when defined(js):
 
   proc parseNativeJson(x: cstring): JSObject {.importjs: "JSON.parse(#)".}
 
+  proc isInteger[T](x: T): bool {.importjs: "Number.isInteger(#)".}
+  proc isSafeInteger[T](x: T): bool {.importjs: "Number.isSafeInteger(#)".}
```
Comment (timotheecour) [inline lib/pure/json.nim]: move to `std/private/jsutils`

**Review by timotheecour — APPROVED** [review]
LGTM after comment addressed

File: lib/pure/json.nim (hunk 1 above)
Comment (ringabout) [reply lib/pure/json.nim]: I'm not sure the parameters, maybe `isSafeInteger` should choose `someInteger`?

Comment (timotheecour) [reply lib/pure/json.nim]: I think it's best to keep it unrestricted, it avoids user doing an extra check (and isSafeInteger allows arbitrary input types)

also, in this or followup PR you can add this `runnableExamples` to std/jsutils :
```nim
proc isInteger[T](x: T): bool {.importjs: "Number.isInteger(#)".}
proc isSafeInteger[T](x: T): bool {.importjs: "Number.isSafeInteger(#)".}

import std/jsffi
assert not "123".toJs.isSafeInteger
assert 123.toJs.isSafeInteger
assert 9007199254740991.toJs.isSafeInteger
assert not 9007199254740992.toJs.isSafeInteger
```

* and in this or future PR, we should also add `Number.MAX_SAFE_INTEGER`

Comment (ringabout) [reply lib/pure/json.nim]: make sense

**Review by timotheecour — APPROVED** [review]

**Review by timotheecour — APPROVED** [review]

File: lib/pure/json.nim (hunk 1 above)
Comment (ringabout) [reply lib/pure/json.nim]: see https://github.com/nim-lang/Nim/pull/17355

**Pull request merged**